### DISCUSSION TITLE
Bmtrain checkpointing mechanism

### DISCUSSION THREAD

**Discussion by Kunlun-Zhu** [OpenBMB/CPM-Live#22] [Open Chats 开放交流]
I realize 'bmtrain'  enable the 'checkpointing' method for optimizing the memory use, will this mechanism helpful for CPM-live training? Though it may cause extra training time required.

**Comment by jayzzhou-thu** [comment]
Hi, thanks for your suggestion! The training of CPM-Live is totally based on BMTrain. The checkpointing method is also used, which can enlarge the training batch size and improve the performance.
